jolly-jellyfish/src/page_maker/models.py

```diff
@@ -1,26 +1,39 @@
-import os
+from datetime import datetime
+from pathlib import Path
 
 from django.conf import settings
 from django.contrib.auth.models import User
 from django.core.validators import validate_image_file_extension
 from django.db import models
 from django.utils import timezone
+from relativefilepathfield.fields import RelativeFilePathField
 
 
 def get_theme_path():
-    return os.path.join(settings.BASE_DIR, 'static')
+    return str(Path(settings.BASE_DIR) / 'static' / 'themes')
 
 
 class Template(models.Model):
-    name = models.CharField(max_length=80)
+    name = models.CharField(max_length=80, unique=True)
     date_created = models.DateTimeField(default=timezone.now)
     author = models.ForeignKey(User, on_delete=models.CASCADE)
     # todo: option to upload own stylesheet rather than select from existing
+    #  potentially upload to user_themes within static not media? -
+    #  makes finding easier below in RelativeFilePathField
     # style_sheet = models.FileField(null=False, blank=False,
     #                      upload_to='themes/', validators=[FileExtensionValidator(['css'])])
 
+    # model.FilePathField() does not have a url attribute,
+    # so accessing the files within a html template is awkward.
+    # Using RelativeFilePathField allows for using django static support easily within the template
     # match uses regex applied to base filename only(.*=any characters, \.=., $=end of string)
-    style_sheet = models.FilePathField(path=get_theme_path, recursive=True, match='.*\.css$')
+    style_sheet = RelativeFilePathField(path=get_theme_path, recursive=True, match='.*\.css$')
```

Comment:
```suggestion
    style_sheet = RelativeFilePathField(path=get_theme_path, recursive=True, match='.+\.css$')
```
Prevent uploads of a file named '.css'